### PR TITLE
[no-ticket][risk=no] First pass at API PR tests

### PR DESCRIPTION
**Pull request by dmohs**
No ticket. Test code changes only.

This deploys the API at pr-X-dot-api-dot-all-of-us-workbench-test.appspot.com and configures the UI deployed at pr-X-dot-all-of-us-workbench-test.appspot.com to point at it, then runs the new E2E tests.

### CODE REVIEW

File: ui/src/environments/environment.pr-site-0.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Environment } from 'environments/environment-type';
+import { testEnvironmentBase } from 'environments/test-env-base';
+
+// This file is used in the deployed test environment
+export const environment: Environment = {
+  ...testEnvironmentBase,
+  allOfUsApiUrl:
+    'https://pr-0-dot-api-dot-all-of-us-workbench-test.appspot.com',
+  displayTag: 'Test',
```
Comment (dmohs) [inline ui/src/environments/environment.pr-site-0.ts:9]: This is useful information: It shows up around the logo. Make it meaningful.

File: .circleci/config.yml [hunk 2]
```diff
@@ -527,6 +533,41 @@ jobs:
...
+            sudo apt install openjdk-8-jdk
+            PATH=/usr/lib/jvm/java-8-openjdk-amd64/bin:"$PATH"
+            gsutil cp gs://all-of-us-workbench-test-credentials/vars.env db
+            set -a; source db/local-vars.env; source db/vars.env; set +a
+            ./gradlew --no-daemon :appengineStage
+
+            PR_NUM="$(echo "$CIRCLE_PULL_REQUEST" | perl -ne "/(\d+)\$/; print \$1")"
+            PR_SITE_NUM="$(expr $PR_NUM % $PR_SITE_COUNT)"
+            gcloud app deploy build/staged-app/app.yaml -q --project=all-of-us-workbench-test \
+              --no-promote --version=pr-$PR_SITE_NUM
```
Comment (dmohs) [inline .circleci/config.yml:570]: I will do a follow-up PR to put these scripts into files rather than leave them inline.

File: e2ev2/src/test-utils.js [hunk 3]
```diff
@@ -78,6 +78,6 @@ const impersonateUser = async (page, username) => {
       .map(x => x.access_token)
       .toPromise()
   await page.evaluate(t => setTestAccessTokenOverride(t), bearerToken)
-  expect(await page.waitForSelector('[data-test-id="signed-in"]', 4e3)).toBeDefined()
+  expect(await page.waitForSelector('[data-test-id="signed-in"]', {timeout: 4e3})).toBeDefined()
```
Comment (dmohs) [inline e2ev2/src/test-utils.js:81]: Unrelated fix.

File: .circleci/config.yml [hunk 4]
```diff
@@ -509,7 +509,13 @@ jobs:
             SA_EMAIL="$(echo "$SA_KEY_JSON" | jq -r .client_email)"
             echo "$SA_KEY_JSON" | gcloud auth activate-service-account "$SA_EMAIL" --key-file=-
 
-            ../e2ev2/src/circle-build-ui.sh
+            export PR_NUM="$(echo "$CIRCLE_PULL_REQUEST" | perl -ne "/(\d+)\$/; print \$1")"
```
Comment (evrii) [inline .circleci/config.yml:512]: Where does $CIRCLE_PULL_REQUEST come from?

**Review by evrii — APPROVED** [review]
On a high level the changes make sense, but if you get a chance, I would love to how this works. I am unsure how to test that this works.

Also, where are the environment files used? I do not see reference to loading them or any of their underlying attributes. Is there somewhere that is reading these configs in and generating environments?

**Comment by dmohs** [comment]
@evrii I have one more cleanup to do, then I'll do another walkthrough.

Our UI uses the environment variable `REACT_APP_ENVIRONMENT` to select an API configuration. That is how the environment files are tied in.

File: .circleci/config.yml (hunk 4 above)
Comment (dmohs) [reply .circleci/config.yml:512]: This, among many others, is set by Circle before starting the job. My preferred way to investigate the environment is to re-run a job with ssh, which is why the Playground job exists. Just run `env` in there to see everything.

Comment (evrii) [reply .circleci/config.yml:512]: I like that. That would be a helpful way to look into this. Thanks for pointing it out.
